File: Go/283. Move Zeroes.go
package main

//goland:noinspection GoUnusedFunction
func moveZeroes(nums []int) {
	length := len(nums)
	zeroSize := 0

	for i := 0; i < length; i++ {
		if nums[i] == 0 {
			zeroSize++
		} else if zeroSize > 0 {
			nums[i-zeroSize] = nums[i]
			nums[i] = 0
		}
	}
}

File: Python/283. Move Zeroes.py
from typing import List


# noinspection PyMethodMayBeStatic
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        length = len(nums)
        zero_size = 0

        for i in range(length):
            if nums[i] == 0:
                zero_size += 1
            elif zero_size > 0:
                nums[i - zero_size] = nums[i]
                nums[i] = 0
